sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/SortOrder.scala

```diff
@@ -61,6 +61,9 @@ case class SortOrder(child: Expression, direction: SortDirection)
   override def sql: String = child.sql + " " + direction.sql
 
   def isAscending: Boolean = direction == Ascending
+
+  def semanticEquals(other: SortOrder): Boolean =
```

Comment:
yea I understand in `EnsureRequirements` we should use `semanticEquals` instead of `==` to compare `SortOrder`, but why we need to implement `samanticEquals` again in `SortOrder`? What's wrong with the default implementation?

I mean, there is no need to "introduce" a `semanticEquals` in `SortOrder`, it already has.